interface/ceed-basis.c

```diff
@@ -607,8 +607,20 @@ int CeedBasisGetCollocatedGrad(CeedBasis basis, CeedScalar *colograd1d) {
 int CeedBasisApply(CeedBasis basis, CeedInt nelem, CeedTransposeMode tmode,
                    CeedEvalMode emode, CeedVector u, CeedVector v) {
   int ierr;
+  CeedInt ulength, vlength, ndof, nqpt;
   if (!basis->Apply) return CeedError(basis->ceed, 1,
-                                        "Backend does not support BasisApply");
+                                      "Backend does not support BasisApply");
+  // check compatibility of topological and geometrical dimensions
+  ierr = CeedBasisGetNumNodes(basis, &ndof); CeedChk(ierr);
+  ierr = CeedBasisGetNumQuadraturePoints(basis, &nqpt); CeedChk(ierr);
+  if (u && v) {
```

Comment:
There must always be a `v`, so we will always have something to check. What about instead of lines 616-618 we declare `ulength=0` on 610, always do `CeedVectoGetLength` on `v`, do `CeedVectorGetLength` on `u` if `u` is not `NULL` and then always run the check on lines 619-622?

Reply:
Done. Thanks!